src/CrudTrait.php

```diff
@@ -191,4 +191,215 @@ public function uploadMultipleFilesToDisk($value, $attribute_name, $disk, $desti
 
         $this->attributes[$attribute_name] = json_encode($attribute_value);
     }
+
+    /**
+     * Handle image upload and DB storage for a image:
+     * - on CREATE
+     *     - stores the image at the destination path
+     *     - generates a name
+     *     - creates image variations
+     *     - stores json object into database with variations and paths
+     * - on UPDATE
+     *     - if the value is null, deletes the file and sets null in the DB
+     *     - if the value is different, stores the different file and updates DB value.
+     *
+     * @param  [type] $value            Value for that column sent from the input.
+     * @param  [type] $attribute_name   Model attribute name (and column in the db).
+     * @param  [type] $disk             Filesystem disk used to store files.
+     * @param  [type] $destination_path Path in disk where to store the files.
+     * @param  [type] $variations       Array of variations and their dimensions
+     */
+    public function uploadImageToDisk($value, $attribute_name, $disk, $destination_path, $variations = null)
+    {
+        if (! $variations || ! is_array($variations)) {
+            $variations = ['original' => null, 'thumb' => [150, 150]];
+        }
+
+        //Needed for the original image
```

Comment:
That comment's confusing - it's needed but it's null? Does that mean the value should be set to the null value maybe?